File: 0036-valid-sudoku/0036-valid-sudoku.py
class Solution:
    def isValidSudoku(self, board: List[List[str]]) -> bool:
        for row in board:
            if not self.checker(row):
                print("failed1", row)
                return False
        
        for i in range(9):
            row = [j[i] for j in board]
            if not self.checker(row):
                print("failed2")
                return False
        
        for i in [0, 3, 6]:
            for j in [0 , 3 , 6]:
                row = [y for x in board[i: i+3] for y in x[j:j+3]]
                if not self.checker(row):
                    print("failed3")
                    return False
        return True

    def checker(self, inlist):
        inputlist = [i for i in inlist if i != '.']
        validset = set([str(i) for i in range(1, 10)])
        repeatcheck = len(set(inputlist)) == len(inputlist)
        validcheck = set(inputlist) <= validset
        return repeatcheck and validcheck
        
                
        

File: 0036-valid-sudoku/0036-valid-sudoku.cpp
class Solution {
public:
    bool isValid3x3( vector<vector<char>>& board, int l, int r ){
        unordered_set<char> st;
        for(int i=l; i<l+3; i++){
            for(int j=r; j<r+3; j++){
                if( board[i][j] != '.' && st.find(board[i][j]) != st.end() ){
                    cout << "3 I returned here i : " << i << " j: " << j << endl;
                    cout << board[i][j];
                    return false;
                }
                st.insert( board[i][j] );
            }
        }
        return true;
    }
    bool isValidSudoku(vector<vector<char>>& board) {
        for(int i=0; i<9; i++){
            unordered_set<char> hst;
            unordered_set<char> st;
            for(int j=0; j<9; j++){
                if( board[i][j] != '.' && st.find(board[i][j]) != st.end() ){
                    cout << "1 I returned here i : " << i << " j: " << j << endl;
                    cout << board[i][j];
                    return false;
                }
                if( board[j][i] != '.' && hst.find(board[j][i]) != hst.end() ){
                    cout << "2 I returned here i : " << i << " j: " << j << endl;
                    cout << board[j][i];
                    return false;
                }
                hst.insert(board[j][i]);
                st.insert(board[i][j]);
            }
        }
        for(int i=0; i<9; i+=3){
            for(int j=0; j<9; j+=3){
                if(!isValid3x3(board, i, j)) return false;
            }
        }
        return true;
    }
};